Lib/socket.py

```diff
@@ -846,6 +855,9 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
             if sock is not None:
                 sock.close()
 
+        if sum_timeouts_so_far >= timeout:
+            break
```

Comment:
Indenting this to be within the `except` block would help reasoning about the code.  The only way this code can be reached is when an exception has happened given that the inside of the try: returns upon success.  Being at the same level makes it more clear that this is not expected to be used in some other condition as well.